source/game_sa/Tasks/TaskTypes/TaskSimplePlayerOnFoot.cpp

```diff
@@ -51,89 +50,59 @@ CTaskSimplePlayerOnFoot::CTaskSimplePlayerOnFoot() : CTaskSimple() {
-        }
-
-        return false;
-    }
-    default:
-        return false;
     }
-
-    // In some specific secnarios clear weapon mode
-    if (   (ped->m_pTargetedObject || (ped->m_pPlayerData->m_nPlayerFlags & 8) != 0 || TheCamera.Using1stPersonWeaponMode())
-        && event
-    ) {
-        if ([&, this] {
-            switch (event->GetEventType()) {
-            case EVENT_DAMAGE: {
-                const auto dmg = static_cast<const CEventDamage*>(event);
-                if (dmg->m_damageResponse.m_bHealthZero && dmg->m_bAddToEventGroup) {
-                    return true;
-                }
-                if (!ped->m_pAttachedTo) {
-                    if (dmg->m_bKnockOffPed) {
-                        return true;
-                    }
-
-                    if (dmg->m_weaponType > WEAPON_LAST_WEAPON && dmg->m_weaponType != WEAPON_UZI_DRIVEBY) {
-                        return true;
-                    }
-                }
-                return false;
-            }
-            case EVENT_IN_WATER:
-                return true;
+    if (abortable) {
+        if ((ped->m_pTargetedObject || (ped->m_pPlayerData->m_nPlayerFlags & 8) != 0 || TheCamera.Using1stPersonWeaponMode()) && event &&
+            (event->GetEventType() == EVENT_DAMAGE || event->GetEventType() == EVENT_IN_WATER)) {
+            if (event->GetEventType() != EVENT_DAMAGE || eventDamage->m_damageResponse.m_bHealthZero && eventDamage->m_bAddToEventGroup ||
+                !ped->m_pAttachedTo && (eventDamage->m_bKnockOffPed || eventDamage->m_weaponType > WEAPON_LAST_WEAPON && eventDamage->m_weaponType != WEAPON_UZI_DRIVEBY)) {
+                TheCamera.ClearPlayerWeaponMode();
+                CWeaponEffects::ClearCrossHair(ped->m_nPedType);
+                CEntity::ClearReference(ped->m_pTargetedObject);
             }
-            return false;
-        }()) {
-            TheCamera.ClearPlayerWeaponMode();
-            CWeaponEffects::ClearCrossHair(ped->m_nPedType);
-            CEntity::ClearReference(ped->m_pTargetedObject);
         }
     }
-
-    // This is only reached if switch (at the beginning) didn't `return false`
-    return true;
+    return abortable;
 }
 
 // 0x688810
 bool CTaskSimplePlayerOnFoot::ProcessPed(CPed* ped) {
-    auto* player = static_cast<CPlayerPed*>(ped);
+    auto player = static_cast<CPlayerPed*>(ped);
 
     if (player->GetPadFromPlayer()) {
         CPedIntelligence* intelligence = player->GetIntelligence();
         bool bPedMoving = player->m_nMoveState >= PEDMOVE_WALK;
-        if (   player->GetActiveWeapon().m_nType == WEAPON_CHAINSAW
-            && intelligence->GetTaskFighting()
-            && intelligence->GetTaskFighting()->m_nCurrentMove == FIGHT_ATTACK_FIGHTIDLE
-        ) {
+        if (player->GetActiveWeapon().m_nType == WEAPON_CHAINSAW && intelligence->GetTaskFighting() && intelligence->GetTaskFighting()->m_nCurrentMove == FIGHT_ATTACK_FIGHTIDLE) {
             bPedMoving = true;
         }
```

Comment:
```suggestion
    if (player->GetActiveWeapon().m_nType == WEAPON_CHAINSAW) {
        if (const auto fightingTask = intelligence->GetTaskFighting()) {
           if (fightingTask->m_nCurrentMove == FIGHT_ATTACK_FIGHTIDLE) {
                   bPedMoving = true;
           }
        }
    }
```
Perhaps that'd be better?

Reply:
Done